lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1005,6 +1007,46 @@ def invocation_step(self, trans, invocation_id, step_id, **kwd):
         )
         return self.__encode_invocation_step(trans, invocation_step)
 
+    def _generate_aws_estimate(self, trans, invocation_id, **kwd):
+        """
+        build AWS cloud estimate
+        """
+        decoded_workflow_invocation_id = self.decode_id(invocation_id)
+        workflow_invocation = self.workflow_manager.get_invocation(trans, decoded_workflow_invocation_id)
+        history = workflow_invocation.history
+
+        aws_dict, metrics, ds_metrics = {}, {}, {}
+        h_contents = self.history_contents_manager.contained(history)
+        for h in h_contents:
+            ds_metrics[h.dataset.id] = {
+                'file_size': int(h.dataset.file_size),
+                'total_size': int(h.dataset.total_size),
+                'uuid': str(h.dataset.uuid)
+            }
+        for i, step in enumerate(workflow_invocation.steps):
+            if step.workflow_step.type == 'tool':
+                for job in step.jobs:
+                    metrics[i] = summarize_job_metrics(trans, job)
+                    for job_input in job.input_datasets:
+                        if hasattr(job_input.dataset, 'dataset_id'):
+                            ds_metrics[job_input.dataset.id].update({'filename': job_input.dataset.name})
+                            metrics[i].append(ds_metrics[job_input.dataset.id])
+                    for job_output in job.output_datasets:
+                        if hasattr(job_output.dataset, 'dataset_id'):
+                            ds_metrics[job_output.dataset.id].update({'filename': job_output.dataset.name})
+                            metrics[i].append(ds_metrics[job_output.dataset.id])
+
+        aws_dict = metrics
+        return aws_dict
+
+    @expose_api
+    def export_aws_estimate(self, trans, invocation_id, **kwd):
```

Comment:
Can you rename this to `invocation_metrics` or similar ? The output may not actually contain aws estimates if that is not configured.